index.js

```diff
@@ -1505,6 +1505,29 @@
   }
   S.pipe = def('pipe', {}, [$.Array($.AnyFunction), a, b], pipe);
 
+  //# pipeK :: Chain m => [(a -> m b), ..., (y -> m z)] -> m a -> m z
+  //.
+  //. Takes an array of functions assumed to be unary which return
+  //. values with a [Chain][], and a value of that Chain, and returns the
+  //. result of applying the sequence of transformations to the initial value.
+  //.
+  //. In general terms, `pipeK` performs left-to-right [Kleisli][] composition
+  //. of an array of functions. `pipeK([f, g, h], x)` is equivalent to
+  //. `chain(h, chain(g, chain(f, x)))`.
+  //.
+  //. ```javascript
+  //. > S.pipeK([S.tail, S.tail, S.head], S.Just([1, 2, 3, 4]))
+  //. Just(3)
+  //. ```
+  function pipeK(fs, x) {
+    return Z.reduce(function(x, f) { return Z.chain(f, x); }, x, fs);
+  }
+  S.pipeK =
+  def('pipeK',
+      {m: [Z.Chain]},
+      [$.Array($.AnyFunction), m(a), m(b)],
```

Comment:
Let's update the type of the first argument to `f($.Any, m($.Any))` as Aldwin suggested. :)

Reply:
👍